meta-mbl-distro/scripts/create-update-payload

```diff
@@ -0,0 +1,191 @@
+your MBL workspace.
+
+The script supports multi-component or single component payloads.
+
+Example invocation:
+
+create-update-payload --bootloader-slots 1 2 --output-tar-path tmp/payload.tar
+
+This will create the update payload containing the bootloader 1 and 2
+components at tmp/payload.tar
+"""
+
+import pathlib
+import re
+import shutil
+import sys
+import tarfile
+import tempfile
+
+from itertools import product
+
+# Set the PYTHONPATH so we can import openembedded scripts/lib modules.
+# This script can only be executed from the
+# meta-mbl-distro/scripts folder in the MBL workspace.
+oe_scripts_path = (
+    pathlib.Path(__file__).parent.parent.parent.parent.resolve()
+    / "openembedded-core"
+    / "scripts"
+)
+
+lib_path = oe_scripts_path / "lib"
+sys.path.insert(0, str(lib_path))
+
+import scriptpath
+import scriptutils
+
+# This is needed so we can import bitbake modules.
+bitbakepath = scriptpath.add_bitbake_lib_path()
+
+import bb.tinfoil
+import argparse_oe
+
+
+def parse_args():
+    """Parse the command line arguments."""
+    parser = argparse_oe.ArgumentParser()
+    parser.add_argument("-f", "--filenames", nargs="+")
+    parser.add_argument("-b", "--bootloader-slots", nargs="+")
+    parser.add_argument(
+        "-o",
+        "--tar-output-path",
+        type=lambda x: pathlib.Path(x).absolute(),
+        default="payload.tar",
+    )
+    parser.add_argument(
+        "-c", "--compression-format", default="gz", choices=["gz", "bz2", "xz"]
+    )
+    args = parser.parse_args()
+
+    if args.filenames is None and args.bootloader_slots is None:
+        raise Exception()
+
+    return args
+
+
+def match_varnames_to(filenames, tinfoil=None):
+    """
+    Match bitbake variable names to the given file names.
+
+    Parse the bitbake config data using the tinfoil API and find the
+    variable which is bound to the file name.
+
+    Return a dict with the variable names as keys and
+    file names as values.
+
+    :param filenames list: List of file names to look for.
+    :param tinfoil Tinfoil: bitbake tinfoil object.
+    """
+    return {
+        cvar.split("_FILENAME")[0]: fname
+        for cvar, fname in product(tinfoil.config_data, filenames)
+        if cvar.startswith("MBL_WKS")
+        and tinfoil.config_data.getVar(cvar) == fname
+    }
+
+
+def match_filenames_to(bootloader_slots, tinfoil=None):
+    """
+    Look up the bitbake variables corresponding to the bootloader slots.
+
+    Parse the bitbake config data using the tinfoil API and match the
+    MBL_WKS_BOOTLOADER variable to the given bootloader slot.
+
+    Return a dict with the variable name as the key and the file name as the
+    value.
+
+    :param bootloader_slots list: List of bootloader slots e.g [1, 2, fs]
+    :param tinfoil Tinfoil: bitbake tinfoil object.
+    """
+    output_map = {}
+    for conf_var, bl_slot in product(tinfoil.config_data, bootloader_slots):
```

Comment:
Again, there will be a lot of `conf_var`s here, right? I think something like the following would be simpler and avoid the massive product iterator.
```Python
for conf_var in tinfoil.config_data:
    m = re.match( r"MBL_WKS_BOOTLOADER_?([A-Z0-9]+)_FILENAME", conf_var)
    if m and m.group(1).lower() in bootloader_slots:
        # ...